pinecone/control/pinecone.py

```diff
@@ -544,6 +544,33 @@ def describe_index(self, name: str):
 
         return IndexModel(description)
 
+    def has_index(self, name: str) -> bool:
+        """Checks if a Pinecone index exists.
+
+        :param name: The name of the index to check for existence.
+        :return: Returns `True` if the index exists, `False` otherwise.
+
+        ### Example Usage
+
+        ```python
+        from pinecone import Pinecone
+
+        client = YourClass()  # Initialize your class instance
+        exists = client.has_index("my_index_name")
+
+        if exists:
+            print("The index exists.")
+        else:
+            print("The index does not exist.")
```

Comment:
I would make this example code like this to be more similar to the style in other examples

```python
import os
from pinecone import Pinecone

api_key = os.environ.get("PINECONE_API_KEY")
pc = Pinecone(api_key=api_key)

if pc.has_index("my_index_name"):
    print("The index exists")
else:
    print("The index does not exist")
```

I think the way this is most likely to be used is in notebooks and examples saying something like this, with no intermediate variables created:

```python
if !pc.has_index("example_index"):
    pc.create_index(... args here)
```